### PR TITLE
PPTP-1076 : OrganisationDetails model refactor and unsupported entity GRS response detection

### PR DESCRIPTION
**Pull request by RyanKellyHMRC**
I've tried to make the models (both the GRS response ones and the internal ones) for all business entity types consistent and have expanded to include `identifiersMatch` as that is required in order to actually do the real work behind this ticket.

The organisationDetails section of registration now looks like this in Mongo (this is for a Sole Trader but they all now share the common `registration` part):

```json
  "organisationDetails" : {
    "organisationType" : "SoleTrader",
    "businessRegisteredAddress" : {
      "addressLine1" : "2 Scala Street",
      "addressLine2" : "Soho",
      "townOrCity" : "London",
      "postCode" : "W1T 2HN",
      "countryCode" : "GB"
    },
    "soleTraderDetails" : {
      "firstName" : "James",
      "lastName" : "Bond",
      "dateOfBirth" : "1977-06-01",
      "nino" : "SE123456C",
      "sautr" : "1234567890",
      "registration" : {
        "identifiersMatch" : true,
        "verificationStatus" : "PASS",
        "registrationStatus" : "REGISTERED",
        "registeredBusinessPartnerId" : "X00000123456789"
      }
    }
```

BE: https://github.com/hmrc/plastic-packaging-tax-registration/pull/107

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [x] Links to dependencies have been included (BE/FE work)
 - [ ] User Acceptance Tests (UAT) were run locally and have passed.

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala [hunk 1]
```diff
@@ -128,7 +129,11 @@ class GrsController @Inject() (
             case SUBSCRIBED => DUPLICATE_SUBSCRIPTION
             case _          => STATUS_OK
           }
-        case None => Future.successful(BUSINESS_IDENTIFICATION_FAILED)
+        case None =>
+          val isSoleTrader =
+            registration.organisationDetails.organisationType.contains(OrgType.SOLE_TRADER)
+          if (!isSoleTrader) {}
+          Future.successful(BUSINESS_IDENTIFICATION_FAILED)
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala:136]: This is the bit I started before deciding to undertake the wider refactor.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala [hunk 2]
```diff
@@ -128,7 +135,19 @@ class GrsController @Inject() (
...
+            case Some(SOLE_TRADER) => Future.successful(BUSINESS_IDENTIFICATION_FAILED)
+            case _ =>
+              registration.organisationDetails.grsRegistration match {
+                case Some(
+                      RegistrationDetails(false, Some("UNCHALLENGED"), "REGISTRATION_NOT_CALLED", _)
+                    ) =>
+                  Future.successful(UNSUPPORTED_ORGANISATION)
+                case _ => Future.successful(BUSINESS_IDENTIFICATION_FAILED)
+              }
+          }
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala:149]: This is the main part of 1076: identify non-sole-trader GRS responses which constitute unsupported organisations.

File: test/spec/PptTestData.scala [hunk 3]
```diff
@@ -122,102 +121,119 @@ trait PptTestData extends RegistrationBuilder with MockAuthAction {
...
-    IncorporationRegistrationDetails(registeredBusinessPartnerId = None,
-                                     registrationStatus = "REGISTRATION_NOT_CALLED"
+  protected val verificationFailedRegistrationDetails: RegistrationDetails =
+    RegistrationDetails(identifiersMatch = true,
+                        verificationStatus = Some("FAIL"),
+                        registrationStatus = "REGISTRATION_NOT_CALLED",
+                        registeredBusinessPartnerId = None
+    )
+
+  protected val identifiersUnmatchedRegistrationDetails: RegistrationDetails =
```
Comment (tmc-ee) [inline test/spec/PptTestData.scala:145]: I'm not very familiar with all of the different Business Verification responses but these details are identical to `unregisteredRegistrationDetails`, is this correct?

Comment (RyanKellyHMRC) [reply test/spec/PptTestData.scala:145]: The `identifiersMatch` flag is different.

Comment (tmc-ee) [reply test/spec/PptTestData.scala:145]: Doh! Apologies and thanks for pointing out the difference :sweat_smile: :+1:

**Review by tmc-ee — APPROVED** [review]

**Review by paulblakehmrc — APPROVED** [review]

**Pull request merged**